feat(invite-email): make OCM invite email optional and improve UI

MahdiBaghbani opened the pull request:
# OCM Invites: Optional Email and UI Improvements

### Email is Now Optional

The backend was already set up to handle NULL emails (thanks, database schema!), but the code was still checking for it. Now you can create invites without email, perfect for when you just want to share a link manually. (in my case test suite)

### Invite Labels

Now there's an optional label field so you can tag invites with something like "Giuseppe from CERN" or "Test invite for demo". Makes searching way easier, especially for link-only invites. 
I just reused the existing `recipient_name` column, so no database migrations needed. until the invite accepted by remote the label lives there, after it would be replaced by the actual name returned by remote

### UI
- Better spacing and layout
- Added a checkbox for "Send invite via email", email fields only appear when you check it
- Clear hint text about what happens if you skip email

The invite details page :
- Just the essential info, no redundant input fields
- Three handy copy buttons in a grid: invite link, token, and base64

Everything is backward compatible, existing invites with email work exactly like before. No breaking changes here as far as I've tested, if you test it on the SUNET test drive it would be great!

### Gallery

#### Pop up changes, email toggle, invalid or not configured smtp error, etc
<img width="2507" height="1399" alt="Screenshot 2025-11-30 112601" src="https://github.com/user-attachments/assets/5aff5254-782c-4029-855b-bbd424e26ad5" />
<img width="2508" height="1397" alt="Screenshot 2025-11-30 112637" src="https://github.com/user-attachments/assets/e1d51681-e574-4edb-9100-53c14b00ace4" />
<img width="2508" height="1398" alt="Screenshot 2025-11-30 112618" src="https://github.com/user-attachments/assets/24972a3d-07d0-4230-b360-4363a7f55627" />
<img width="2506" height="1398" alt="Screenshot 2025-11-30 112649" src="https://github.com/user-attachments/assets/448bd08f-daf3-43ad-bdff-dd1eb73528e6" />

#### Creating a test invite with no email
<img width="2510" height="1399" alt="Screenshot 2025-11-30 112709" src="https://github.com/user-attachments/assets/95321579-3439-45dc-8d0c-f1cf75434935" />
<img width="2508" height="1399" alt="Screenshot 2025-11-30 112724" src="https://github.com/user-attachments/assets/165ff7aa-310f-4ea8-bbd5-3d7446e4daa0" />

mickenordin reviewed (approved):
I like it!

redblom commented:
@MahdiBaghbani - can you rebase and squash maybe ?

redblom commented:
@MahdiBaghbani - tested locally and came up with the recommendations below. 
Looks great btw. :+1: 
ps. I'm pushing for merges of this and the required server code. Hopefully soon now.

 #### New Invite:
 
The email_optional flag:
  - if set to false the email_optional checkbox is not visible. But if we don't specify an email in the invite form the current message still reads: Please enter an email address or uncheck "Send invite via email".
     It should just read: Please enter an email address.

The email:
  - Optional personal message is missing in the email sent (apparently it's not set anymore in the frontend).

#### Accept invite forms (popup & manual):
 - Explanatory messages on these two forms should be aligned. They have diverged.
   Proposal for both forms:
    - Header: Accept invite to exchange contact info.
    - Explanatory message: 
       Accepting will add the inviter to your contacts list and in return, your contact info will be sent to the inviter.
       From there on you can start sharing data with each other.
 
#### Accept invite form (manual):
 - the helper and label texts of the input field should match.

#### All invites list:
 - An invite that has been send by email, and can be resend ('Resend mail' button visible), also shows the 'Copy invite link&code' buttons. I'm not sure this is useful here.
 - For invites that have not been sent by email the 'Sent to' field reads: No email (link-only)
    I think we can remove (link-only). Or indeed, the whole 'Sent to' row.

#### Settings:
 - maybe create commands for the new config settings

MahdiBaghbani commented:
OK! glad you liked it.
I have a bit of a problem over here; I'll try to sort things out early next week and ask for review again.
BTW, you mean pushing to merge this https://github.com/nextcloud/contacts/pull/4417 ? It would be cool!

redblom commented:
> OK! glad you liked it. I have bit of problems over here, I'll try to sort things out early next week and ask for review again. 

If you can manage, great, else just give a shout and I can try to do it.

> BTW, you mean pushing to merge this [nextcloud#4417](https://github.com/nextcloud/contacts/pull/4417) ? It would be cool!

Yes! It belongs in there :)

MahdiBaghbani committed:
fix(ocm): align invite flow contracts

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani commented:
@redblom I know I'm taking more than a week to wrap things up 😃 sorry to keep you waiting!

I tried implementing some of @artonge reviews as well. 

This is almost ready but needs a bit more edits

redblom committed:
feat(OCM-Invites): Implement exchanging cloud IDs through OCM invitation workflow
see https://github.com/cs3org/OCM-API/blob/v1.2.1/IETF-RFC.md

Features:
 - Button to invite remote users to exchange cloudIDs.
 - Button to manually accept invite to exchange cloudIDs.
 - WAYF page allowing the receiver of the invite to open and accept the invitation.
 - Listing of open invitations.
 - Option to resend, revoke open invitations.

Signed-off-by: Antoon P. <antoon.prins@surf.nl>
Co-authored-by: Micke Nordin <kano@sunet.se>
Co-authored-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

redblom commented on lib/Listener/FederatedInviteAcceptedListener.php:
```diff
@@ -9,66 +9,71 @@
+ * Listens to the federated invite accepted event.
+ * Catching the event should lead to the creation of the new remote contact
+ * from the invite, in the inviter's address book.
+ *
```
This does not work anymore. The event dispatched is `OCMEndpointRequestEvent`. See `https://github.com/sara-nl/nextcloud-contacts/blob/invite-for-cloudid-exchange/lib/Listener/FederatedInviteAcceptedListener.php`.
I delegated acceptance and contact creation to `FederatedInvitesService`.

MahdiBaghbani replied:
I have been working on this

Also as per OCM spec the email in payload is required not optional so that has to be fixed too

redblom commented on lib/Command/DisableOcmInvites.php:
A suggestion from NC was to remove these commands that set only one parameter and people can simply use `occ config:app:set {appid} {param} --value {value} --type {type}` (https://github.com/nextcloud/contacts/pull/4417#discussion_r3087368455).
I think it makes sense. 
Maybe we can put everything into `OcmInvitesConfig` ?

MahdiBaghbani commented on lib/Command/OcmInvitesConfig.php:
```diff
@@ -39,19 +38,35 @@ public function __construct(
 	#[\Override]
 	protected function configure(): void {
+		$supportedOptions = implode(', ', $this->getSupportedOptions());
+		$help = <<<HELP
```
Decide to keep this or remove this, I figured maybe a help would be helpful 😄

redblom replied:
Yes. What would be nice is to have an explanation for each setting, what it's for.
Either in this text or per setting.
I'm trying to figure out if that's possible.

MahdiBaghbani committed:
update(ocm): adopt external invitations labels

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani committed:
update(nav): move invitation actions to sidebar

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani committed:
style(ocm): restyle invite form email section

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani committed:
update(ocm): refresh invitation details panel

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani committed:
update(ocm): drop cc-sender from backend

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani committed:
update(ocm): drop cc-sender from admin ui

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani committed:
feat(ocm): add collapsible email preview to form

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani committed:
update(ocm): polish invite accept modal layout

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani committed:
update(ocm): polish wayf provider picker layout

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani committed:
fix(ocm): count pending invites in nav badge

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>

MahdiBaghbani committed:
style(ocm): normalize invite list copy spacing

Signed-off-by: Mahdi Baghbani <mahdi-baghbani@azadehafzar.io>
